Compile failure with base-4.11 / inaccurate version bounds

`dates-0.2.2.1` advertises being compatible with `base >=4 && < 5`, but this is falsifiable by e.g. `base-4.11.1.0` which results in the build failure below:

```
Build profile: -w ghc-8.4.3 -O1
In order, the following will be built (use -v for more details):
 - dates-0.2.2.1 (lib) (first run)
Configuring library for dates-0.2.2.1..
Preprocessing library for dates-0.2.2.1..
Building library for dates-0.2.2.1..
[1 of 4] Compiling Data.Dates.Internal ( Data/Dates/Internal.hs, /tmp/dates-0.2.2.1/dist-newstyle/build/x86_64-linux/ghc-8.4.3/dates-0.2.2.1/build/Data/Dates/Internal.o )
[2 of 4] Compiling Data.Dates.Types ( Data/Dates/Types.hs, /tmp/dates-0.2.2.1/dist-newstyle/build/x86_64-linux/ghc-8.4.3/dates-0.2.2.1/build/Data/Dates/Types.o )

Data/Dates/Types.hs:62:10: error:
    • No instance for (Semigroup DateTime)
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Monoid DateTime’
   |
62 | instance Monoid DateTime where
   |          ^^^^^^^^^^^^^^^
```

There's no immediate action required as I have (in my function as Hackage Trustee) already performed a metadata revision for you on Hackage, see http://hackage.haskell.org/package/dates-0.2.2.1/revisions/

In case you intend to add support for base-4.11 the information at https://prime.haskell.org/wiki/Libraries/Proposals/SemigroupMonoid#RecommendedVariant may be useful.

If you have any questions, please don't hesitate to ask me!

--- a/dates.cabal
+++ b/dates.cabal
@@ -41,11 +41,11 @@ library
                        Data.Dates.Formats,
                        Data.Dates.Internal
 
-  build-depends:       base >=4 && < 4.12,
+  build-depends:       base >=4.9 && < 4.13,
                        base-unicode-symbols ==0.2.*,
-                       time >= 1.4,
-                       parsec >=3.1,
-                       syb >=0.3.7
+                       time >= 1.4 && < 1.9,
+                       parsec ==3.1.*,
+                       syb >=0.3.7 && < 0.8
 
   GHC-Options:         -O2
 

